cpp/include/algorithms.h

```diff
@@ -170,3 +172,20 @@ gdf_error gdf_louvain(gdf_graph *graph,
                       void *final_modularity,
                       void *num_level,
                       gdf_column *louvain_parts);
+
+/**
+ * @brief Compute connected components. 
+ * The weak version was imported from cuML.
+ * This implementation comes from [1] and solves component labeling problem in
+ * parallel on CSR-indexes based upon the vertex degree and adjacency graph.
+ *
+ * [1] Hawick, K.A et al, 2010. "Parallel graph component labelling with GPUs and CUDA"
+ *
+ 
+ * @param graph input graph; assumed undirected for weakly CC [in]
+ * @param labels gdf_column for the output labels [out]
+ * @param connectivity_type 0=WEAK; 1=STRONG
+ */
+gdf_error gdf_connected_components(gdf_graph *graph,
```

Comment:
Do we have any rule on ordering function parameters? Here, we have [input], [output], [input], and this looks unorthodox to me; but other functions in this header also mixes input & output parameters. I guess we should have input parameters first then output parameters (or output parameters first then input parameters following CUDA's convention, but I guess the former is more popular and somewhat more consistent with other functions in the same header file). And we should enforce const correctness. Here, are we modifying graph inside gdf_connected_components? If not, it should be marked as const.

Reply:
Things like that should be agreed upon _before_ integration. _After_ integration should probably be addressed in bulk in a special refactoring PR.

Reply:
OK, no complaint about this at this moment.